app/models/teacher.rb

```diff
@@ -292,21 +284,21 @@ def self.csv_export
     end
   end
 
-  private
-  def ensure_unique_personal_email
-    # We want to ensure that both email and personal_email are unique across
-    # BOTH columns (i.e. the email only appears once in the combined set)
-    # However, it's perfectly valid for personal_email to be nil
-    # TODO: This really suggests email needs to be its own table...
-    teacher_with_email = Teacher.where.not(id: self.id).exists?(["email = :value OR personal_email = :value", { value: self.email }])
-    if teacher_with_email
-      errors.add(:email, "must be unique across both email and personal_email columns")
-    end
-    return unless self.personal_email.present?
+  def email
+    # Default return primary email
+    primary_email
+  end
 
-    teacher_personal_email = Teacher.where.not(id: self.id).exists?(["email = :value OR personal_email = :value", { value: self.personal_email }])
-    if teacher_personal_email
-      errors.add(:personal_email, "must be unique across both email and personal_email columns")
-    end
+  def primary_email
+    email_addresses.find_by(primary: true)&.email
```

Comment:
It looks like this line was reviewed on an outdated code. My revised code since last time was `email_addresses.find_by(primary: true)&.email || self[:email]`. Here I use `self[:email]` instead of `email` directly due to a naming conflicting (`email` is used as a method name for some other reasons.)